### PR TITLE
feat(content-sharing): Display shared link settings

### PR DESCRIPTION
**Pull request by sahiga**
Enable toggling between the [Shared Link Settings Modal](http://opensource.box.com/box-ui-elements/storybook/?path=/story/features-sharedlinksettingsmodal--basic) and the Unified Share Modal in the Content Sharing Element. The Shared Link Settings Modal will appear when the user clicks "Link settings"; the Unified Share Modal will appear by default, and when the Shared Link Settings Modal is closed.

content-sharing-slsm

### CODE REVIEW

File: src/features/shared-link-settings-modal/SharedLinkSettingsModal.js [hunk 1]
```diff
@@ -303,15 +303,15 @@ class SharedLinkSettingsModal extends Component {
 
         return (
             message && (
-                <p className="link-settings-modal-notice">
+                <div className="link-settings-modal-notice">
```
Comment (swfree) [inline src/features/shared-link-settings-modal/SharedLinkSettingsModal.js:306]: It looks like this change has removed some bottom margin, if you compare your screenshot to: http://opensource.box.com/box-ui-elements/storybook/?path=/story/features-sharedlinksettingsmodal--basic
Is that the intended change?

**Comment by manishPh** [comment]
LGTM apart from the margin issue mentioned above.

File: src/features/shared-link-settings-modal/SharedLinkSettingsModal.js (hunk 1 above)
Comment (sahiga) [reply src/features/shared-link-settings-modal/SharedLinkSettingsModal.js:306]: Good catch! I updated the CSS rule and the image snapshot.

File: src/elements/content-sharing/ContentSharing.js [hunk 2]
```diff
@@ -54,6 +55,7 @@ function ContentSharing({ apiHost, displayInModal, itemID, itemType, language, t
         changeSharedLinkPermissionLevel,
         setChangeSharedLinkPermissionLevel,
     ] = React.useState<null | SharedLinkUpdateFnType>(null);
+    const [currentView, setCurrentView] = React.useState<string>(CONTENT_SHARING_VIEWS.UNIFIED_SHARE_MODAL);
```
[the remark this answers is not in the document]
Comment (sahiga) [reply src/elements/content-sharing/ContentSharing.js:58]: That would be ideal. Right now, the two forms live in separate modals, so this is the only way to toggle between them 😞  I hope that we can eventually move SharedLinkSettings out of the modal, but that would probably require a refactor on the WebApp side as well.

[the remark this answers is not in the document]
Comment (sahiga) [reply src/elements/content-sharing/ContentSharing.js:58]: I think we'd always want to show the USM by default. SharedLinkSettings only appears when you click "Link settings" in the USM.

**Comment by sahiga** [comment]
@ConradJChan @swfree @manishPh Thank you for reviewing! I updated the PR. Can you take another look?

**Pull request merged**